File: 0359-logger-rate-limiter/0359-logger-rate-limiter.java
class Logger {
    private HashMap<String, Integer> cache = new HashMap<>();

    public boolean shouldPrintMessage(int timestamp, String message) {
        var before = cache.getOrDefault(message, -987654321);
        if (before + 10 <= timestamp) {
            cache.put(message, timestamp);
            return true;
        }
        return false;
    }
}

/**
 * Your Logger object will be instantiated and called as such:
 * Logger obj = new Logger();
 * boolean param_1 = obj.shouldPrintMessage(timestamp,message);
 */

File: 0359-logger-rate-limiter/0359-logger-rate-limiter.py
class Logger:

    def __init__(self):
        self.cache = dict()

    def shouldPrintMessage(self, timestamp: int, message: str) -> bool:
        before = self.cache.get(message, -987654321)
        if before + 10 <= timestamp:
            self.cache[message] = timestamp
            return True
        return False
        


# Your Logger object will be instantiated and called as such:
# obj = Logger()
# param_1 = obj.shouldPrintMessage(timestamp,message)